core/src/main/java/org/apache/iceberg/MetadataUpdate.java

```diff
@@ -23,11 +23,17 @@
 import java.util.Map;
 import java.util.Set;
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+import org.apache.iceberg.view.ViewMetadata;
+import org.apache.iceberg.view.ViewVersion;
 
-/** Represents a change to table metadata. */
+/** Represents a change to table/view metadata. */
 public interface MetadataUpdate extends Serializable {
   void applyTo(TableMetadata.Builder metadataBuilder);
 
+  default void applyTo(ViewMetadata.Builder viewMetadataBuilder) {
+    throw new UnsupportedOperationException("Applying updates to view metadata is not supported");
```

Comment:
What about `"Cannot apply update %s to a view"` and use `this.getClass().getSimpleName()`? I think that's a bit more clear.